### ISSUE TITLE
I3D提取

### ISSUE THREAD

**Issue by modadundun** [Aaron-Pu/PEL4VAD#20]
作者您好，我用您推荐的I3D提取方法，但是提取出来的序列长度有问题，想了解您提取SH特征时候 extraction_fps\stack_size\step_size这三个是如何设置的呢，十分感谢！！

**Comment by yujiangpu20** [comment]
Thank you for the question. Since we use 16 frames as a step size, not all videos can get an integer number of segments. Therefore, after extracting the features, we calibrate the ground-truth file to match the segment length. Please refer to https://github.com/yujiangpu20/PEL4VAD/blob/master/list/sh/rectify_gt.py

**Comment by modadundun** [comment]
Thank you so much! You are truly the fastest responder I’ve ever met. I will be citing you in my upcoming paper and recommending you to others! Regarding the code, if I use your features and labels as the original ones, and my extracted features as the target features, should I generate the corresponding labels based on this code? Or is there a separate file for the original features and labels?

**Comment by yujiangpu20** [comment]
Yes, if you extract the features yourself, you should first use the script to align the labels with your features. However, I recommend using the features and the GT file I provided, so you can run the code directly without any modifications.

**Comment by modadundun** [comment]
> Yes, if you extract the features yourself, you should first use the script to align the labels with your features. However, I recommend using the features and the GT file I provided, so you can run the code directly without any modifications.

Thank you so much for your response! Since I wanted to obtain the original (non-10-crop) features, I extracted them myself but found that the sequence length of each feature did not match the ones from your features. So, I plan to regenerate the ground truth (GT) file based on the code you provided. However, I think this method may introduce some errors. Do you happen to have the original (non-10-crop) features available? I would be extremely grateful if you could provide them! Your guidance and help have been immensely beneficial to me!!

**Comment by yujiangpu20** [comment]
Actually the ten-crop features are four-corner and center crop with their flipped counterparts. And you may find the feature is named like "01_0015_0.npy", where the last index indicates different crops. Here '0' means the center crop. So if you only want to use a single crop for training and testing, you can modify the dataloader to load only that crop. Also you need to modify the inference code since we calculated the average of 10-crop logits. Hope it helps~

**Comment by modadundun** [comment]
What I meant was not to use 10-crop but to directly extract I3D features without cropping. However, I appreciate your suggestion to modify the code, and I will make the changes. Once again, thank you. Your responses are always helpful.
